### DISCUSSION TITLE
Express returns Runtime.Unknown

### DISCUSSION THREAD

**Discussion by TheDummi** [vercel/community#3865] [Help]
### Summary

So I'm trying to put a REST API on my domain but when I deploy it to vercel it says serverless function has crashed and the logs tell me that it's a Runtime.Unknown error.

My app is in root as I don't want it to be in an api or public directory. 

### Steps to Reproduce

I can't really provide replications as the code uses an env that holds a key to my database.

**Comment by amyegan** [comment]
Hey @TheDummi. We have a guide with lots of info about how to deploy an Express app with Vercel: https://vercel.com/guides/using-express-with-vercel

**Reply by TheDummi** [reply]
I've read it multiple times, I just can't figure out what I'm doing wrong.

**Reply by TheDummi** [reply]
The things I might do wrong and can't seem to get confirmation on:
> - Does it have to be in a public or api directory? If so does this mean all my url endpoints will start with this prefix?
> - Does it have to export the app, and if so can this be done in esm rather than cjs as they have different types of exporting

**Reply by jsshigham** [reply]
did you figure this out? I am having the same issue
